lib/modules/datasource/types.ts

```diff
@@ -70,10 +70,16 @@ export interface ReleaseResult {
   replacementVersion?: string;
 }
 
+export type MetadataResult = Pick<
+  ReleaseResult,
+  'changelogUrl' | 'dependencyUrl' | 'homepage' | 'sourceUrl'
+>;
```

Comment:
I think that we should flip this around:
- Introduce a new `PackageMetadata` interface
- Make `ReleaseResult` extend it, and add `releases` (every other field should be possible for this new function to return).

BTW we could also rename `ReleaseResult` to be e.g. `PackageReleases` or `PackageWithReleases`, if you think that name is more understandable.

Reply:
good idea.

Reply:
Yeah, good idea. I am just going to keep `ReleaseResult` for now, because we already have well established the `Result` suffix for return types.